pkg/azure/validation/validation.go

```diff
@@ -0,0 +1,288 @@
+/*
+SPDX-FileCopyrightText: 2020 SAP SE or an SAP affiliate company and Gardener contributors
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+// Package validation - validation is used to validate cloud specific ProviderSpec
+package validation
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/gardener/machine-controller-manager/pkg/apis/machine/v1alpha1"
+	"github.com/gardener/machine-controller-manager/pkg/util/provider/machinecodes/codes"
+	"github.com/gardener/machine-controller-manager/pkg/util/provider/machinecodes/status"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	"github.com/gardener/machine-controller-manager-provider-azure/pkg/azure/api"
+	"github.com/gardener/machine-controller-manager-provider-azure/pkg/azure/utils"
+)
+
+const providerAzure = "Azure"
+
+// ValidateMachineClassProvider checks if the Provider in MachineClass is Azure.
+// If it is not then it will return an error indicating that this provider implementation cannot fulfill the request.
+func ValidateMachineClassProvider(mcc *v1alpha1.MachineClass) error {
+	if mcc.Provider != providerAzure {
+		return status.Error(codes.InvalidArgument, fmt.Sprintf("Request for provider %s cannot be fulfilled. Only %s provider is supported.", mcc.Provider, providerAzure))
+	}
+	return nil
+}
+
+// ValidateProviderSpec validates the api.AzureProviderSpec.
+func ValidateProviderSpec(spec api.AzureProviderSpec) field.ErrorList {
+	var allErrs field.ErrorList
+	specPath := field.NewPath("providerSpec")
+
+	if len(strings.TrimSpace(spec.Location)) == 0 {
```

Comment:
why not use `utils.IsEmptyString()` here as well, and other similar places?

Reply:
Thanks all other places have it sans 2 places where i have changed it now as well